src/librustc_back/lib.rs

```diff
@@ -169,3 +169,48 @@ impl ToJson for RelroLevel {
         }
     }
 }
+
+#[derive(Clone, Copy, Debug, PartialEq, Hash, RustcEncodable, RustcDecodable)]
+pub enum VaListKind {
+    CharPtr,
+    VoidPtr,
+    AArch64Abi,
+    PowerPcAbi,
+    X86_64Abi,
+    None
```

Comment:
Is this variant just a placeholder for targets which haven’t specified their `VaList` ABI? Removing this variant and picking *some* default would would end up removing the `bug` branch from trans.

Reply:
The void or char pointer could be a sensible default.

Reply:
> Is this variant just a placeholder for targets which haven’t specified their VaList ABI? 

I don't believe we support one that doesn't specify a VaList ABI, and I can't think of one off the top of my head that hasn't specified their VaList ABI, so unless someone else can think of a reason to keep the `none` variant, I'll remove it and use `void-ptr` or `char-ptr` as the default.